### ISSUE TITLE
T6963

### ISSUE THREAD

**Issue by olikraus** [olikraus/u8glib#327]
Originally reported on Google Code with ID 328

```
Hi,

I´m usin a WG240128B lcd with T6963 chip. I connect de lcd to an arduino mega2560 and
arduino soft 1.6.1 like in the proyect.

when power on de lcd flash once and nothing more.
The proyect compile wheel. 

The lcd don´t have pins A0 and CS, have C/D(comand/data read/write) and CE (chip enable).
I conect A0 like CE and CS like C/D.

Please help.
```

Reported by `goosze` on 2015-03-12 17:05:37

<hr>
- _Attachment: [wg240128b.pdf](https://storage.googleapis.com/google-code-attachments/u8glib/issue-328/comment-0/wg240128b.pdf)_

**Comment by pivo88** [comment]
Hi guys,
I'd like to reanimate this issue somehow. I'm playing around now with Arduino Leonardo and my old-one JM12864B (128x64) LCD display with T6963C controller. I've opened "Hello World" example, configured pinout and u8glib constructor, compiled and upload and nothing just happened.

I made a connection by wireing like this:
![lcd jm12864b connection to arduino leonardo](https://user-images.githubusercontent.com/36489719/36229220-d1df886c-11d6-11e8-8d99-f36b5e50e188.jpg)

You can find My LCD display Spec here: http://www.jm.pl/karty/JM12864BSPEC.pdf or here: https://kentec.net/pdf/lcd/12864b.pdf

u8glib constructor:
//U8GLIB_T6963_128X64 u8g(D0, D1, D2, D3, D4, D5, D6, D7, cs, a0, wr, rd, reset);
  U8GLIB_T6963_128X64 u8g( 6,  7,  8,  9, 10, 11, 12, 13,  3,  4,  1,  2, 5);

FS pin is connected to GND.
I'm not sure what should I do with pin 4 Vee...

Compilation goes fine, upload aswell but the display stays blank.

Do you have any idea what goes wrong?

**Comment by pivo88** [comment]
Ok, sorry I didn't notice u8g is not supported any more and there is new u8g2.
But I've made also a try and it still doesn't work :(

**Comment by olikraus** [comment]
The U8g2 constructor is different. You must connect Rd to 5v, if I remember correctly. See also the U8g2 FAQ.

Vee usually requires -10v but this depends on your board also.

**Comment by pivo88** [comment]
> The U8g2 constructor is different. You must connect Rd to 5v, if I remember correctly. See also the U8g2 FAQ.

Yes, I noticed that :) I've adapted constructor to U8g2 version and connected RD to +5V, but this is not the case this time. later in the evening I will try to use another Breadboard and will make connection again.

**Comment by jakubtrzpiot** [comment]
Were you able to make it work? I have the same problem. I know you posted 5 years ago but if you could share your solution I'd be grateful.

**Comment by olikraus** [comment]
U8g2 supports the T6963 without problems

**Comment by jakubtrzpiot** [comment]
My display is POWERTIP PG-12864F 
I'm new to electronics and I wasn't able to make it work though I wired everything according to my displays documentation I found online and used this constructor: U8G2_T6963_128X64_1_8080 u8g2(U8G2_R0, 8, 9, 10, 11, 4, 5, 6, 7, /*enable=*/ 17, /*cs=*/ 14, /*dc=*/ 15, /*reset=*/ 16); // Connect RD with +5V, FS0 and FS1 with GND.
I'd appreciate if you could just point me in the right direction. Thanks!
![image](https://user-images.githubusercontent.com/92888924/224833835-bcd60d86-2d63-48f3-8679-8b9343d0e704.png)

**Comment by pivo88** [comment]
Hi there,
I didn't develop my project any more after that and didn't solve the problem.
Maybe I'll come back to this as I'm still having both Arduino and LCD ;)

All the best in your development!

**Comment by olikraus** [comment]
Did you add some negative voltage to v0?

**Comment by jakubtrzpiot** [comment]
@pivo88 
Hi and thanks for replying!
Too bad you had to abandon your project :( but I hope you'll be able to make it work some day.

**Comment by jakubtrzpiot** [comment]
@olikraus
No, I didn't was I supposed to? I thought I only had to use negative voltage for the backlight. Should I use the inverter output to change the contrast with a pot?

**Comment by jakubtrzpiot** [comment]
Excuse my lack of competence in electronics. I have V0 connected to the potentiometer with +5V and GND. I've also used the wrong term in the previous comment (I thought an inverter inverts voltage from +5 to -5 but in actuality, it goes DC to AC which is used for the EL backlight in my display). I'll try to get my hands on a charge pump to produce negative voltage for the V0 pin and see how it goes. Thanks for your help and for creating this awesome library!

**Comment by olikraus** [comment]
... the negative voltage is required for the LCD itself

**Comment by jakubtrzpiot** [comment]
I used some old printer power supply that has -12V rail and now I can actually adjust the contrast. I still have the same problem as op where the text isn't displayed on the screen but when I connect the reset pin to the ground I get a single horizontal line that moves down each time I put the pin to ground. (I suspect it has something to do with the refresh rate?)

**Comment by jakubtrzpiot** [comment]
Nothing I tried made any difference so I switched to a display with KS0108 controller and everything worked right away with u8g2 lib!

**Comment by olikraus** [comment]
hmmm

**Comment by lomovtsevds** [comment]
I have a similar display based on the T6963 chip and similar issue with that one line.
First of all, I recommend setting the contrast. You should use a tester to check the input voltage as required(5V). After checking the voltage, the next step for you is to assemble the power circuit (5V)Vdd - Vo - Vee (-16V) ,with a potentiometer and then enable the power. After that (and one horizontal line), rotate the potentiometer towards Vss (counter clockwise in my schematic above) until you see the screen getting darker. You don't need to make it very dark, just enough to be slightly visible.
After that, using a library (I used U8g2), you can easily make the display work! BTW the chip is sensitive to timing. I had to decrease my MCU's (STM32F4xx) clock speed (from 186 MHz to 42 MHz) to get a stable picture on the screen. Its worked on GPIO and FSMC.
